main.go

```diff
@@ -115,6 +116,39 @@ type ObjRef struct {
 	Offloaded uint
 }
 
+// updateAutoretrieveIndex ticks every tickInterval and checks for new information to add to autoretrieve
+// If so, it updates the filecoin index with the new CIDs, saying they are present on autoretrieve
+// With that, clients using bitswap can query autoretrieve servers using bitswap and get data from estuary
+func (s *Server) updateAutoretrieveIndex(tickInterval time.Duration, quit chan struct{}) error {
+	var autoretrieves []Autoretrieve
+	var lastTickTime time.Time
+	ticker := time.NewTicker(tickInterval)
+
+	for {
+		lastTickTime = time.Now().UTC().Add(-tickInterval)
```

Comment:
If I understand the intent, by the time a tick is read of the ticker channel, the `autoretrieves` wont exactly be peers whose `last_connection` is slightly above `- tickInterval`, it will be almost double (peers we last saw 720*2 minutes ago). Better to do this logic inside the `ticker.C` case